File: src/types/dto/location.ts
export type LocationDto = {
  id: string;
  path: string;
  name: string;
  sortIndex: number;
  enabled: boolean;
  isDefault: boolean;
};

export type PathDiskSpaceDto = {
  path: string;
  totalBytes: number;
  availableBytes: number;
};

export type AddLocationPayload = {
  path: string;
  name?: string | null;
};

export type UpdateLocationPayload = {
  id: string;
  path?: string | null;
  name?: string | null;
  sortIndex?: number | null;
  enabled?: boolean | null;
  isDefault?: boolean | null;
};

export type LocationOrderEntry = {
  id: string;
  sortIndex: number;
};

export type SetFavoritePayload = {
  id: string;
  favorite: boolean;
};

export type ScanResultDto = {
  projectsDiscovered: number;
  projectsUpserted: number;
  projectsPruned: number;
  dirsSkippedErrors: number;
  monoreposExpanded: number;
  workspaceWarnings: number;
};

File: src-tauri/src/models/location.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LocationDto {
    pub id: String,
    pub path: String,
    pub name: String,
    pub sort_index: i32,
    pub enabled: bool,
    pub is_default: bool,
}


#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PathDiskSpaceDto {
    pub path: String,
    pub total_bytes: u64,
    pub available_bytes: u64,
}


